pkg/deploy-manager/common.go

```diff
@@ -105,32 +160,79 @@ func (t *DeployManager) GetDeploymentImage(name string) (string, error) {
 }
 
 // WaitForPVCBound waits for a pvc with a given name and namespace to reach BOUND phase
-func (t *DeployManager) WaitForPVCBound(pvcName string, pvcNamespace string) {
-	gomega.Eventually(func() error {
-		pvc, err := t.k8sClient.CoreV1().PersistentVolumeClaims(pvcNamespace).Get(pvcName, metav1.GetOptions{})
-		if err != nil {
-			return err
+func (t *DeployManager) WaitForPVCBound(pvc *k8sv1.PersistentVolumeClaim, namespace string) error {
+	pvc, err := t.k8sClient.CoreV1().PersistentVolumeClaims(namespace).Create(pvc)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	lastReason := ""
+	timeout := 100 * time.Second
+	interval := 1 * time.Second
```

Comment:
Instead of retrying every second for 100 seconds, can we just do retry every 5 seconds for 10 times?
This is mentioned in K8s guidelines and practices too. Creating a PVC and binding it shouldn't take 100 seconds in a properly functioning system. If it is, something needs to be fixed. Adding more timeout just gives more window for bugs to sneak in.